modules/core/osgi-service-tracker-collections/src/main/java/com/liferay/osgi/service/tracker/collections/internal/ServiceTrackerUtil.java

```diff
@@ -14,6 +14,8 @@
 
 package com.liferay.osgi.service.tracker.collections.internal;
 
+import com.liferay.portal.kernel.util.StringBundler;
```

Comment:
This is the cause of the failure. osgi-service-tracker-collections does not have kernel dependencies.
You will have to use petra-string's StringBundler here, and add petra-string's dependencies.
As a matter of fact, @tomwang2011 will be replacing globally to move all kernel StringBundler's usage to petra-string's StringBundler.
You can either update all usages to that, or just do it for osgi-service-tracker-collections, leaving the rest to @tomwang2011